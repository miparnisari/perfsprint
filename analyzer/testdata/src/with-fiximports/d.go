package a

import (
	"fmt" // want "fiximports: Fix imports"
	"strings"
)

func loopconcat4() {
	// only fmt removed
	s := ""
	for i := 0; i < 10; i++ {
		s += "toto" // want "concat-loop: string concatenation in a loop"
	}
	println("Hello, World!", strings.Fields(fmt.Sprintf("%s", s))) // want "string-format: fmt.Sprintf can be replaced with just using the string"
}
