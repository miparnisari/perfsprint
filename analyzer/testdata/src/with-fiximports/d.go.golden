package a

import (
	// want "fiximports: Fix imports"
	"strings"
)

func loopconcat4() {
	// only fmt removed
	s := ""
	var sSb11 strings.Builder
	for i := 0; i < 10; i++ {
		sSb11.WriteString("toto") // want "concat-loop: string concatenation in a loop"
	}
	s += sSb11.String()
	println("Hello, World!", strings.Fields(s)) // want "string-format: fmt.Sprintf can be replaced with just using the string"
}
